### PR TITLE
[FLINK-2872] [Documentation] Update the documentation for Scala part to add ExecutionEnvironment.readFileOfPrimitives.

### PR DESCRIPTION
**Pull request by hsaputra**
Add the missing Scala part for ExecutionEnvironment.readFileOfPrimitives API doc in the
programming guide.

### CODE REVIEW

File: docs/apis/programming_guide.md [hunk 1]
```diff
@@ -1721,7 +1721,8 @@ File-based:
   Returns a DataSet of tuples or POJOs. Supports the basic java types and their Value counterparts as field
   types.
 
-- `readFileOfPrimitives(path, Class)` / `PrimitiveInputFormat` - Parses files of new-line (or another char sequence) delimited primitive data types such as `String` or `Integer`. 
+- `readFileOfPrimitives(path, Class)` / `PrimitiveInputFormat` - Parses files of new-line (or another char sequence)
```
Comment (fhueske) [inline docs/apis/programming_guide.md]: Can you add the delimiter parameter to the Java version as well to have both APIs in sync?

**Comment by fhueske** [comment]
Thanks for fixing the documentation. The Java API offers also a variant that allows to specify a delimiter. It might be good to extend the signature of the Java method to have Scala and Java in sync.

**Comment by hsaputra** [comment]
@fhueske , yeah good idea, will add the variant of the signature in the Java part

**Comment by hsaputra** [comment]
@fhueske , I have updated the PR with the Java variant info.

**Comment by hsaputra** [comment]
Will merge this sometime today if no more comments.

**Pull request closed without merging**